### ISSUE TITLE
version consisting of four numbers

### ISSUE THREAD

**Issue by llipavsky** [GitTools/GitVersion#1582]
Hi, I'd like to use GitVersion for continuous integration of our (existing) product.

However the versioning of our product is slightly different to the one expected by GitVersion

we have 4 parts version: 7.1.12.5
(major.servicepack.release.buildnumber). 
Is it possible to somehow configure the GitVersion to support this versioning?

In fact the major (and also servicepack) versions are changed so rarely, that it would be perfectly fine for me to treat them as constants and change GitVersion configuration so it only affects last three (or two), but I want the "generated" version to be complete 4-number

I know that I can simply "prepend 7. to the version in some wrapper scripts", but I want e.g. branches/tags to contain complete version (full 4 numbers) and I am afraid that this could cause trouble to GitVersion

So, is something like that possible?

Thnaks

**Comment by LordMike** [comment]
You'll be battling a lot with GitVersion, but we managed to do it with this config:

```yaml
assembly-versioning-scheme: MajorMinorPatchTag
assembly-file-versioning-scheme: MajorMinorPatchTag
mode: ContinuousDeployment
continuous-delivery-fallback-tag: ''
branches:
  master:
    increment: None
  develop:
    increment: Patch
  hotfix:
    tag: useBranchName
    increment: None
  release:
    increment: None
  pull-request:
    increment: None
  support:
    increment: None
ignore:
  sha: []
```

Essentially we configure GitVersion to use 4 version numbers for Assembly and File versions. Then, we configure  it to not increment the three first version numbers in any case (as we control it using tags).

We _cannot_ with this include the full 4-version tag in Git, as GitVersion will start interpreting that in all sorts of ways.. But then, we don't need the 4-version tags (as the build number in our case doesn't matter .. yet).

**Comment by llipavsky** [comment]
Thanks, looks promising. I'll try that

**Comment by dwfrancis** [comment]
We ran into a similar situation - our major/minor version numbers are locked do to certification contracts. We've configured GitVersion to ignore the first number. Bbranch `release/7.1.12.5` and tag `v7.1.12.5` will be both parse `1.12.5` as the major.minor.patch - we then add the `7.` on the assembly version formats.

```
tag-prefix: 'v?7\.'
assembly-file-versioning-format: '7.{Major}.{Minor}.{Patch}'
assembly-versioning-format: '7.{Major}.{Minor}.{Patch}'
increment: Patch
```

There are a few additional scripts to add the `7.` to the build, package and display versions.
